processor/src/main/java/org/mapstruct/ap/internal/model/common/TypeFactory.java

```diff
@@ -261,7 +261,7 @@ else if ( type.isPrimitive() ) {
      */
     public ExecutableType getMethodType(TypeElement includingType, ExecutableElement method) {
         DeclaredType asType = (DeclaredType) replaceTypeElementIfNecessary( elementUtils, includingType ).asType();
-        TypeMirror asMemberOf = typeUtils.asMemberOf( asType, method );
+        TypeMirror asMemberOf = SpecificCompilerWorkarounds.asMemberOf( asType, method, typeUtils, elementUtils );
```

Comment:
Should we add a CS rule to make sure that nowhere in our code base the original `isMemberOf()` is invoked? IIRC, they provide a way to fail the build if specific methods are invoked.